### PR TITLE
fix(kit): fix Firefox issue with input right padding for case when icon or cleaner on the right

### LINKED ISSUE #680: [BUG] Invisible caret in Firefox at inputs edge
# 🐞 Bug report

image

Pull request is in place, created this issue to properly track :)

### PR DESCRIPTION
**Pull request by k-tar**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [ ] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

Closes #680

## What is the current behavior?

In Firefox, padding for input is being ignored until text's width less than input's width. Some text become unable to see under the cleaner/icons.
screenshot

## What is the new behavior?

Replaced paddings with invisible borders. 

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x ] No

## Other information

Checked that fix works and does not break anything:
- Chrome, Firefox, Safari – on MacOS;
- Chrome – on Android;
- Safari – on iOS.

### CODE REVIEW

File: projects/core/styles/mixins/textfield.less [hunk 1]
```diff
@@ -51,18 +51,20 @@
...
     text-transform: inherit;
 
     :host[data-tui-host-size='s'] & {
-        padding: 0 0.5rem 0 0.75rem;
+        border-width: 0 5rem 0 0.75rem;
     }
 
     :host[data-tui-host-size='l'] & {
-        padding: 0 1rem;
+        border-width: 0 16px;
```
Comment (waterplea) [inline projects/core/styles/mixins/textfield.less:67]: Let's keep rems here and instead change the function to rem itself

File: projects/core/components/primitive-textfield/primitive-textfield.template.html [hunk 2]
```diff
@@ -17,8 +17,8 @@
         automation-id="tui-primitive-textfield__native-input"
         class="input"
         [class.input_hidden]="inputHidden"
-        [style.paddingLeft.px]="paddingLeft"
-        [style.paddingRight.px]="paddingRight"
+        [style.borderLeftWidth.px]="borderLeft"
```
Comment (waterplea) [inline projects/core/components/primitive-textfield/primitive-textfield.template.html:20]: Please change all `.px` to `.rem`

File: projects/core/utils/miscellaneous/get-border.ts [hunk 3]
```diff
@@ -1,32 +1,32 @@
-const PADDING_SMALL = 12;
-const PADDING_LARGE = 16;
+const BORDER_SMALL = 12;
```
Comment (waterplea) [inline projects/core/utils/miscellaneous/get-border.ts:1]: Let's change all these to REMs:
```js
const BORDER_SMALL = 0.75;
const BORDER_LARGE = 1;
const ITEM_SIZE = 1.5;
const CONTENT_SIZE = 2.5;
```

File: projects/core/components/primitive-textfield/primitive-textfield.style.less [hunk 4]
```diff
@@ -50,6 +50,9 @@
 }
 
 .input {
+    border-style: solid;
```
Comment (waterplea) [inline projects/core/components/primitive-textfield/primitive-textfield.style.less:53]: Can we maybe change this to a shorthand `border: 0 solid transparent;`?

**Review by waterplea — CHANGES_REQUESTED** [review]
Thank you for this PR! Please take a look at my comments

**Comment by waterplea** [comment]
@k-tar we plan to start regular releases on Mondays, if you fix comments today we can get this into next release ;)

**Comment by k-tar** [comment]
@waterplea Everything resolved

File: projects/core/components/primitive-textfield/primitive-textfield.template.html [hunk 5]
```diff
@@ -40,8 +40,8 @@
         *ngIf="inputHidden"
         automation-id="tui-primitive-textfield__value"
         class="input input_template"
-        [style.paddingLeft.px]="paddingLeft"
-        [style.borderRightWidth.px]="paddingRight"
+        [style.borderLeftWidth.px]="borderLeft"
```
Comment (waterplea) [inline projects/core/components/primitive-textfield/primitive-textfield.template.html:43]: These two still have `px`

File: projects/core/styles/mixins/textfield.less [hunk 6]
```diff
@@ -51,18 +51,20 @@
         .autofill(dark);
     }
 
-    padding: 0 0.75rem;
+    padding: 0;
+    border-left-width: 0.75rem;
```
Comment (waterplea) [inline projects/core/styles/mixins/textfield.less:55]: Can it be `border-width: 0 0.75rem`?

**Review by waterplea — CHANGES_REQUESTED** [review]
A few little comments + tests need to be updated to use REMs. Please check the final result on Safari and iOS, I don't have those near me these days and they tend to break things :)

**Comment by k-tar** [comment]
@waterplea 

Fixed everything and found one more required change for icon paddings (in primitive-textfield.component.ts).

Safari and iOS was checked (iOS only in Xcode Simulator, I haven't a real device, but Simulator was correct all time I use it for CSS checks), works fine and everything as before, except behaviour related to the issue.

**Review by waterplea — APPROVED** [review]

**Pull request merged**